odoo/addons/base/models/res_currency.py

```diff
@@ -284,10 +287,40 @@ def is_zero(self, amount):
         self.ensure_one()
         return tools.float_is_zero(amount, precision_rounding=self.rounding)
 
+    @property
+    def CACHED_FIELDS(self) -> tuple[str, ...]:
+        """ Return fields to cache for the active currencies.
+        Please promise all these fields don't depend on other models and context
+        and are not translated.
+        """
+        return (
+            'name', 'symbol', 'position', 'decimal_places', 'active',
+        )
+
+    def _fetch_field(self, field):
+        if any(self._ids) and field.name in self.CACHED_FIELDS:
+            self._check_field_access(field, 'read')
+            data = self._cached_data()[field.name]
+            field._insert_cache(self.browse(data), data.values())
+            if all(record_id in data for record_id in self.ids):
+                self.check_access('read')
+                return
+        super()._fetch_field(field)
+
+    @tools.ormcache(cache='stable')
+    def _cached_data(self) -> frozendict:
+        """ Cached values for active currencies. """
+        fnames = self.CACHED_FIELDS
+        records = self.sudo().with_context({}).search_fetch([('active', '=', True)], fnames, order='name')
+        return frozendict({
+            fname: frozendict(zip(records.ids, map(self._fields[fname]._get_cache(records.env).__getitem__, records.ids)))
+            for fname in fnames
+        })
```

Comment:
There is some redundancy in this data structure: the `records.ids` are used as keys for all fields.  And `fetch()` does not even take advantage from the field mapping `{id: value}`, as it basically "unzip" it to call `field._insert_cache()`.

I suggest to put all field values in tuples, including `records.ids`:
```py
fields_ = [self._fields[fname] for fname in fnames]
return records._ids, frozendict({
    field: tuple(map(field._get_cache(records.env).__getitem__, records._ids))
    for field in fields_
})
```